### DISCUSSION TITLE
Retrieve type of Creditor information from fact

### DISCUSSION THREAD

**Discussion by ahoward-ch** [manusimidt/py-xbrl#98] [Q&A]
I have an XBRL file with 2 types of creditors: `Creditors: amounts falling due within one year` and `Creditors: amounts falling due after more than one year`. But the `fact` for each only contains `Creditors` in the name and I cant see any other attributes present which hold the information. 

If I convert the instance to JSON, then the dictionary for that fact does contain the key/value pair of `'MaturitiesOrExpirationPeriodsDimension': 'WithinOneYear'` for the first, and `'MaturitiesOrExpirationPeriodsDimension': 'AfterOneYear'` for the second. I really need to access that value when scanning through the instance fact by fact, so I can append the `'WithinOneYear'`/`'AfterOneYear'` to the concept name. 

Any ideas how?

**Comment by manusimidt** [comment]
From your description, the problem is unfortunately not exactly clear to me. When a concept is defined in the taxonomy, you can specify whether it has a credit or debit character.  Temporal information is normally added via the context. Could you maybe share some of the xml code (the facts and corresponding contexts)?  Maybe I can help you then.
